src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInfo.cs

```diff
@@ -26,6 +26,12 @@ public abstract partial class MethodInfo : MethodBase
         public virtual Delegate CreateDelegate(Type delegateType) { throw new NotSupportedException(SR.NotSupported_SubclassOverride); }
         public virtual Delegate CreateDelegate(Type delegateType, object? target) { throw new NotSupportedException(SR.NotSupported_SubclassOverride); }
 
+        /// <summary>Creates a delegate of the given type 'T' from this method.</summary>
+        public T CreateDelegate<T>() where T : System.Delegate => (T)CreateDelegate(typeof(T));
```

Comment:
This can be just `where T : Delegate`. We usually do not use fully qualifies names in `src`. The fully qualified type names are only used in `ref`.